pkg/cassandra/encryption.go

```diff
@@ -0,0 +1,234 @@
+}
+
+func ReadEncryptionStoresSecrets(ctx context.Context, klusterKey types.NamespacedName, template *DatacenterConfig, remoteClient client.Client, logger logr.Logger) (encryption.EncryptionStoresPasswords, error) {
+	encryptionStoresPasswords := encryption.EncryptionStoresPasswords{}
+	if ClientEncryptionEnabled(template) {
+		if err := checkMandatoryEncryptionFields(template.ClientEncryptionStores); err != nil {
+			return encryptionStoresPasswords, err
+		}
+		logger.Info("Client to node encryption is enabled, reading client encryption stores secrets")
+		// Read client keystore password
+		if password, err := ReadEncryptionStorePassword(ctx, klusterKey.Namespace, remoteClient, template.ClientEncryptionStores.KeystoreSecretRef.Name, encryption.StoreNameKeystore); err != nil {
+			return encryptionStoresPasswords, err
+		} else {
+			encryptionStoresPasswords.ClientKeystorePassword = password
+		}
+
+		encryptionStoresPasswords.ClientTruststorePassword = ""
+		if template.ClientEncryptionStores.TruststoreSecretRef != nil {
+			if password, err := ReadEncryptionStorePassword(ctx, klusterKey.Namespace, remoteClient, template.ClientEncryptionStores.TruststoreSecretRef.Name, encryption.StoreNameTruststore); err != nil {
+				return encryptionStoresPasswords, err
+			} else {
+				encryptionStoresPasswords.ClientTruststorePassword = password
+			}
+		}
+	}
+
+	if ServerEncryptionEnabled(template) {
+		logger.Info("Internode encryption is enabled, reading server encryption stores secrets")
+		// Read server keystore password
+		if err := checkMandatoryEncryptionFields(template.ServerEncryptionStores); err != nil {
+			return encryptionStoresPasswords, err
+		}
+
+		if password, err := ReadEncryptionStorePassword(ctx, klusterKey.Namespace, remoteClient, template.ServerEncryptionStores.KeystoreSecretRef.Name, encryption.StoreNameKeystore); err != nil {
+			return encryptionStoresPasswords, err
+		} else {
+			logger.Info(fmt.Sprintf("Read keystore password %s", password))
+			encryptionStoresPasswords.ServerKeystorePassword = password
+		}
+
+		// Read server truststore password
+		encryptionStoresPasswords.ServerTruststorePassword = ""
+		if template.ClientEncryptionStores.TruststoreSecretRef != nil {
+			if password, err := ReadEncryptionStorePassword(ctx, klusterKey.Namespace, remoteClient, template.ServerEncryptionStores.TruststoreSecretRef.Name, encryption.StoreNameTruststore); err != nil {
+				return encryptionStoresPasswords, err
+			} else {
+				logger.Info(fmt.Sprintf("Read truststore password %s", password))
+				encryptionStoresPasswords.ServerTruststorePassword = password
+			}
+		}
+	}
+
+	return encryptionStoresPasswords, nil
+}
+
+func ReadEncryptionStorePassword(ctx context.Context, namespace string, remoteClient client.Client, secretName string, storeName encryption.StoreName) (string, error) {
```

Comment:
FYI I generally think that functions like this belong into controller code/package. Because all they do is take a client and use it to read Kubernetes objects. Same for `ReadEncryptionStoresSecrets`. These 2 functions have little to do with Cassandra configuration.

Reply:
Noted.
It would involve quite a few changes here so I'd like to skip for this time and do this as part of a future refactoring.

Reply:
Agreed 👍